### ISSUE TITLE
RPC Loadwallet crash

### ISSUE THREAD

**Issue by PastaPastaPasta** [dashpay/dash#3521]
steps to reproduce: `./src/qt/dash-qt --testnet` then open console and `loadwallet wallet.dat`

See https://github.com/dashpay/dash/pull/3462 for more information. This is a known issue which is fixed by https://github.com/bitcoin/bitcoin/pull/14552. This PR will be backported as soon as possible (not possible atm due to other backports 'blocking' it). 

At this point, this is a known problem with Dash v0.16

**Comment by thephez** [comment]
Still had this issue with e20dc9f37

```
dash-qt: wallet/db.cpp:236: BerkeleyEnvironment::VerifyResult BerkeleyEnvironment::Verify(const string&, BerkeleyEnvironment::recoverFunc_type, std::__cxx11::string&): Assertion `mapFileUseCount.count(strFile) == 0' failed.
Posix Signal: Aborted
   0#: (0x5631F7AF3935) stl_vector.h:466    - std::vector<unsigned long, std::allocator<unsigned long> >::operator=(std::vector<unsigned long, std::allocator<unsigned long> >&&)
   1#: (0x5631F7AF3935) stacktraces.cpp:801 - HandlePosixSignal
   2#: (0x7FA05964E980) <unknown-file>      - ???
   3#: (0x7FA057B20FB7) <unknown-file>      - ???
   4#: (0x7FA057B22921) <unknown-file>      - ???
   5#: (0x7FA057B1248A) <unknown-file>      - ???
   6#: (0x7FA057B12502) <unknown-file>      - ???
   7#: (0x5631F7B6A4A8) mutex:111           - std::recursive_mutex::lock()
   8#: (0x5631F7B6A4A8) sync.h:59           - AnnotatedMixin<std::recursive_mutex>::lock()
   9#: (0x5631F7B6A4A8) std_mutex.h:267     - std::unique_lock<CCriticalSection>::lock()
  10#: (0x5631F7B6A4A8) sync.h:128          - CCriticalBlock::Enter(char const*, char const*, int)
  11#: (0x5631F7B6A4A8) sync.h:149          - CCriticalBlock::CCriticalBlock(CCriticalSection&, char const*, char const*, int, bool)
  12#: (0x5631F7B6A4A8) db.cpp:235          - BerkeleyEnvironment::Verify(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool (*)(boost::filesystem::path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&), std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
  13#: (0x5631F7B6A71D) db.cpp:353          - BerkeleyBatch::VerifyDatabaseFile(boost::filesystem::path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool (*)(boost::filesystem::path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&))
  14#: (0x5631F7BC9195) wallet.cpp:5029     - CWallet::Verify(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
  15#: (0x5631F7B7D060) basic_string.h:211  - std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_is_local() const
  16#: (0x5631F7B7D060) basic_string.h:220  - std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_dispose()
  17#: (0x5631F7B7D060) basic_string.h:647  - std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()
  18#: (0x5631F7B7D060) rpcwallet.cpp:3256  - loadwallet(JSONRPCRequest const&)
  19#: (0x5631F78D9891) server.cpp:633      - CRPCTable::execute(JSONRPCRequest const&) const
  20#: (0x5631F7AD2781) basic_string.h:211  - std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_is_local() const
  21#: (0x5631F7AD2781) basic_string.h:220  - std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_dispose()
  22#: (0x5631F7AD2781) basic_string.h:647  - std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()
  23#: (0x5631F7AD2781) server.h:37         - JSONRPCRequest::~JSONRPCRequest()
  24#: (0x5631F7AD2781) node.cpp:341        - executeRpc
  25#: (0x5631F75500CA) univalue.h:20       - UniValue::operator=(UniValue&&)
  26#: (0x5631F75500CA) rpcconsole.cpp:318  - RPCConsole::RPCParseCommandLine(interfaces::Node*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const*)
  27#: (0x5631F7550A73) rpcconsole.cpp:425  - RPCExecutor::request(QString const&, QString const&)
  28#: (0x5631F75511CD) rpcconsole.moc:81   - RPCExecutor::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)
Aborted (core dumped)

```

**Comment by UdjinM6** [comment]
Fixed by #4081
